macros/stitch/base/stitch_bing_keyword_performance.sql

```diff
@@ -9,50 +9,69 @@
 with source as (
 
     select * from {{var('keyword_performance_report_table')}}
-    
+
 ),
 
 renamed as (
 
     select
-    
+
         "__SDC_PRIMARY_KEY" as keyword_performance_report_id,
 
-        convert_timezone('UTC', timeperiod)::timestamp_ntz::date 
-            as campaign_date,
-            
+        {% if target.type == 'bigquery' %}
+
+            datetime(timeperiod, 'UTC') as campaign_date,
+
+        {% else %}
+
+            convert_timezone('UTC', timeperiod)::timestamp_ntz::date
+                as campaign_date,
+
+        {% endif %}
+
         accountid as account_id,
         adgroupid as ad_group_id,
         adgroupname as ad_group_name,
         adid as ad_id,
         campaignid as campaign_id,
         campaignname as campaign_name,
         keywordid as keyword_id,
-        
+
         campaignstatus as campaign_status,
-        
+
         replace(
-            coalesce(destinationurl, '') || coalesce(finalurl, ''),
+            concat(coalesce(destinationurl, ''), coalesce(finalurl, '')),
             '%20', ' '
             ) as url,
-        
+
         clicks,
         impressions,
         spend,
-        
-        rank() over (
-            partition by timeperiod::date 
-            order by _sdc_report_datetime desc
+
+        {% if target.type == 'bigquery' %}
```

Comment:
same thing here - we can remove the if/else if we make a macro that will convert the timeperiod to a date in a cross-database way. Do you happen to know what datatype `timeperiod` is in this model on BigQuery?